CCOL-1344 add setting to limit messages logging (#178)
flipp-oss/deimos: docs/CONFIGURATION.md

configurations.

## Base Configuration
Config name| Default                     |Description
-----------|-----------------------------|-----------
logger| `Logger.new(STDOUT)`        |The logger that Deimos will use.

> payload_log| `nil`                       |Determines the amount of data passed to the log.</br>`nil` - pass keys and payloads of each message</br>`:keys` - to pass only keys</br>`:nothing` - to pass only total count of messages 

Reviewer: Let's make this more explicit and set the default to `:full` instead of `nil` (internally we can treat them the same). Also `:nothing` isn't accurate since we're still logging the message count - let's change this to `:count`.

> payload_log| `:full`                     |Determines the amount of data passed to the log.</br>`:full` - pass keys and payloads of each message</br>`:keys` - to pass only keys</br>`:count` - to pass only total count of messages 

Reviewer: ```suggestion
payload_log| `:full`                     |Determines how much data is logged per payload.</br>`:full` - all keys and payloads are logged.</br>`:keys` - only keys are logged.</br>`:count` - only the total count of messages are logged.
```
